tests/validation/tests/v3_api/test_create_ha.py

```diff
@@ -140,7 +140,7 @@ def valid_response():
         assert False, "check the logs in console for details"
 
     print_kubeconfig(kubeconfig_path)
-    if RANCHER_HA_HARDENED and RANCHER_LOCAL_CLUSTER_TYPE == "RKE":
+    if (RANCHER_HA_HARDENED.upper() == "TRUE" and RANCHER_LOCAL_CLUSTER_TYPE == "RKE") or (RANCHER_HA_HARDENED.upper() == "TRUE" and RANCHER_LOCAL_CLUSTER_TYPE == "K3S"):
```

Comment:
The steps in prepare_hardened_cluster after that, doesn't apply for k3s cluster. It applies only for RKE1 cluster.

And the condition that should be here is
```
if (RANCHER_HA_HARDENED.upper() == "TRUE" and RANCHER_LOCAL_CLUSTER_TYPE == "RKE") and RANCHER_HA_KUBECONFIG == ""(<blank>) and RANCHER_HA_HOSTNAME=="" (<blank>)
```
Since the hardening of cluster should happen only when harden flag is set, and the local cluster is RKE1, and when user doesn't input values for kubeconfig and hostname. As the default in this job is always
- RANCHER_HA_HARDENED = True
- RANCHER_LOCAL_CLUSTER_TYPE = RKE1